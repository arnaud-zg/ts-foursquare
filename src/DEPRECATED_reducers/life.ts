import { createReducer } from 'typesafe-actions'
import { NLifeState } from '../../types'
import {
  getAccessTokenAsync,
  putCredentials,
  setOAuth2,
  TLifeAction,
} from '../DEPRECATED_actions'

export const initialState: NLifeState.IState = {
  status: false,
  credentials: {
    clientId: '',
    clientSecret: '',
  },
}

export const lifeReducer = createReducer<NLifeState.IState, TLifeAction>(
  initialState
)
  .handleAction(
    getAccessTokenAsync.success,
    (state, action): NLifeState.IState => ({
      ...state,
      status: true,
      credentials: {
        ...state.credentials,
        ...action.payload,
      },
    })
  )
  .handleAction(
    setOAuth2,
    (state, action): NLifeState.IState => ({
      ...state,
      status: true,
      credentials: {
        ...state.credentials,
        ...action.payload,
      },
    })
  )
  .handleAction(
    putCredentials,
    (state, action): NLifeState.IState => ({
      ...state,
      status: true,
      credentials: {
        ...state.credentials,
        ...action.payload,
      },
    })
  )
