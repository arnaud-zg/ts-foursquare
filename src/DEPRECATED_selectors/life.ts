import { NStore } from '../../types'
import {
  EAuthApiDefaultParameters,
  EAuthApiPathnames,
  EResponseType,
} from '../constants/api'
import { getLocationHref } from '../utils/url'

export const lifeSelector = (state: NStore.IState) => state.life

export const lifeStatusSelector = (state: NStore.IState) =>
  lifeSelector(state).status

export const lifeCredentialsSelector = (state: NStore.IState) =>
  lifeSelector(state).credentials

export const lifeAccessTokenSelector = (state: NStore.IState) =>
  lifeCredentialsSelector(state).accessToken

export const lifeOAuth2UrlSelector = (state: NStore.IState) => {
  const { redirectUri, clientId } = lifeCredentialsSelector(state)

  return getLocationHref({
    origin: EAuthApiDefaultParameters.ORIGIN,
    pathname: EAuthApiPathnames.AUTHENTICATE,
    param: {
      client_id: clientId,
      redirect_uri: redirectUri,
      response_type: EResponseType.CODE,
    },
  })
}
