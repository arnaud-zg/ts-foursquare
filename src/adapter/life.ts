import { NAuth, NOAuth2 } from '../../types'

export const adaptGetAccessToken = (
  payload: NOAuth2.IResponse
): NAuth.IAuth => ({
  accessToken: payload.access_token,
})
