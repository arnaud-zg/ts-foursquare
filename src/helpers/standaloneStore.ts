import { AnyAction, StoreCreator } from 'redux'

type Listener = (action: AnyAction | null, state: any) => void

export class StandaloneStore {
  private listeners: Array<Listener>
  private store: ReturnType<StoreCreator>
  private lastAction: AnyAction | null

  constructor({ store }: { store: ReturnType<StoreCreator> }) {
    this.lastAction = null
    this.listeners = []
    this.store = store

    this.store.subscribe(this.storeUpdate)
  }

  dispatchAction = (action: AnyAction) => {
    this.lastAction = action
    this.store.dispatch(action)
  }

  storeUpdate = () => {
    const state = this.store.getState()

    if (this.listeners.length) {
      this.listeners.forEach(listener => {
        listener(this.lastAction, state)
      })
    }
  }

  subscribe(listener: Listener) {
    this.listeners.push(listener)
  }
}
