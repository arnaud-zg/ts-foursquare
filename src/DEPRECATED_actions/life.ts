import { ActionType, createAction, createAsyncAction } from 'typesafe-actions'
import { NRequest, NResolve } from '../../types'
import { adaptGetAccessToken } from '../adapter'

export const setOAuth2 = createAction('SET_OAUTH2')<NResolve.IOAuth2>()

export const getAccessTokenAsync = createAsyncAction(
  'GET_ACCESS_TOKEN_REQUEST',
  'GET_ACCESS_TOKEN_SUCCESS',
  'GET_ACCESS_TOKEN_FAILURE',
  'GET_ACCESS_TOKEN_CANCEL'
)<
  NRequest.IOAuth2AccessTokenPayload,
  ReturnType<typeof adaptGetAccessToken>,
  Error,
  string
>()

export const putCredentials = createAction('PUT_CREDENTIALS')<
  NResolve.ICredentialsPayload
>()

export const lifeActions = {
  setOAuth2,
  getAccessTokenAsync,
  putCredentials,
}

export type TLifeAction = ActionType<typeof lifeActions>
