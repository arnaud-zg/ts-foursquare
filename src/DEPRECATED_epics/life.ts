import { Epic } from 'redux-observable'
import { of } from 'rxjs/internal/observable/of'
import { catchError, filter, map, switchMap, takeUntil } from 'rxjs/operators'
import { isActionOf } from 'typesafe-actions'
import { NStore } from '../../types'
import { getAccessTokenAsync, TRootAction } from '../DEPRECATED_actions'
import { adaptGetAccessToken } from '../adapter'
import { getObservableAccessToken } from '../services'

export const getAccessTokenEpic: Epic<
  TRootAction,
  TRootAction,
  NStore.IState
> = (action$, state$) =>
  action$.pipe(
    filter(isActionOf(getAccessTokenAsync.request)),
    switchMap(action =>
      getObservableAccessToken({ action, state$ }).pipe(
        map(adaptGetAccessToken),
        map(getAccessTokenAsync.success),
        catchError(err => of(getAccessTokenAsync.failure(err))),
        takeUntil(action$.pipe(filter(isActionOf(getAccessTokenAsync.cancel))))
      )
    )
  )
