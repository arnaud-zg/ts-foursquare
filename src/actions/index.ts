import { TLifeAction } from './life'
import { TVenuesAction } from './venues'

export type TRootActionType = TVenuesAction | TLifeAction

export type TRootAction = TVenuesAction | TLifeAction

export * from './life'
export * from './venues'
