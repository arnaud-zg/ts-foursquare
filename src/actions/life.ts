import { NLife } from '../../types/life'
import { lifeReducer } from 'reducers/life'

export enum EActionRequest {
  GET_LIFE = 'GET_LIFE',
}

export const getLife = (): Promise<NLife.ILife> =>
  new Promise(resolve => {
    resolve(
      lifeReducer({
        type: EActionRequest.GET_LIFE,
        payload: {
          life: true,
        },
      })
    )
  })
