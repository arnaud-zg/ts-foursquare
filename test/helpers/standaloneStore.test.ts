import { getType } from 'typesafe-actions'
import { StandaloneStore } from '../../src/helpers/standaloneStore'
import { configureStore } from '../../src/reducers'
import { putCredentials } from '../../src/actions/life'

describe('helpers/StandaloneStore', () => {
  const store = configureStore()
  it(`should work without any listener`, () => {
    const standaloneStore = new StandaloneStore({ store })

    standaloneStore.dispatchAction(
      putCredentials({
        clientId: '123',
        clientSecret: '456',
      })
    )
  })

  it(`should make a snapshot of store after of action: putCredentials`, done => {
    const standaloneStore = new StandaloneStore({ store })

    standaloneStore.subscribe((action, state) => {
      expect(state).toMatchSnapshot()

      if (action && action.type === getType(putCredentials)) {
        done()
      }
    })

    standaloneStore.dispatchAction(
      putCredentials({
        clientId: '123',
        clientSecret: '456',
      })
    )
  })
})
