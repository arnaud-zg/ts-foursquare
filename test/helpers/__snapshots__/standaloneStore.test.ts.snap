// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers/StandaloneStore should make a snapshot of store after of action: putCredentials 1`] = `
Object {
  "life": Object {
    "credentials": Object {
      "clientId": "123",
      "clientSecret": "123",
    },
    "status": true,
  },
  "status": Object {},
  "venues": Object {
    "entities": Object {},
    "recommendedPlaces": Array [],
  },
}
`;
